source/extensions/filters/network/common/redis/fault.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <chrono>
+#include <string>
+
+#include "envoy/common/pure.h"
+
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+/**
+ * Fault Type.
+ */
+enum class FaultType { Delay, Error };
+
+class FaultManager {
+public:
+  virtual ~FaultManager() = default;
+
+  /**
+   * Get fault type and delay given a Redis command.
+   * @param command supplies the Redis command string.
+   */
+  virtual absl::optional<std::pair<FaultType, std::chrono::milliseconds>>
+  getFaultForCommand(std::string command) const PURE;
+};
+
+using FaultManagerPtr = std::shared_ptr<FaultManager>;
```

Comment:
naming convention for shared_ptr are FaultManagerSharedPtr.  Another question is why is this a shared_ptr, isn't it completely owned by the command splitter and not exposed outside of it?

Reply:
Good call will convert to std::unique_ptr